File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var longestConsecutive = function(nums) {
    const setNums = new Set(nums);
    
    maxSeqLen = 0;
    
    for (let num of nums) {
        if (setNums.has(num - 1)) continue;
        
        let currSeqLen = 1;
        while (setNums.has(num + 1)) {
            currSeqLen++;
            num++;
        }
        if (currSeqLen > maxSeqLen) maxSeqLen = currSeqLen;
    }
    
    return maxSeqLen;
};

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, nums: list[int]) -> int:
        set_nums = set(nums)
        longest = 0

        for num in nums:
            if (num - 1) in set_nums:
                continue
            candidate = 0
            while num in set_nums:
                num += 1
                candidate += 1
            longest = max(longest, candidate)

        return longest
